panther_driver/src/battery_node.py

```diff
@@ -2,153 +2,128 @@
 
 import math
 import yaml
-from numpy import NaN
 
 import rospy
-from sensor_msgs.msg import BatteryState
-from panther_msgs.msg import BatteryDriver
-
 
-class BatteryNode:
-    def __init__(self, name):
-        rospy.init_node(name)
+from sensor_msgs.msg import BatteryState
+from panther_msgs.msg import DriverState
 
-        self._battery_driv_sub = rospy.Subscriber(
-            'battery_driver', BatteryDriver, self._battery_driv_callback
-        )
+from constants import A, B, R0, R1, U_SUPPLY
 
-        self.V_driv_front = NaN
-        self.V_driv_rear = NaN
-        self.I_driv_front = NaN
-        self.I_driv_rear = NaN
 
-        # Battery
-        self._battery_publisher = rospy.Publisher('battery', BatteryState, queue_size=1)
-        self._battery1_publisher = rospy.Publisher('battery1', BatteryState, queue_size=1)
-        self._battery2_publisher = rospy.Publisher('battery2', BatteryState, queue_size=1)
+class PantherBatteryNode:
+    def __init__(self, name) -> None:
+        rospy.init_node(name, anonymous=False)
 
-        if rospy.has_param('~measurements_file'):
-            measurements_file = rospy.get_param('~measurements_file')
-        else:
-            rospy.logerr(
-                f'[{rospy.get_name()}] measurements_file not defined, can not start collecting ADC measurements'
-            )
-            return
+        measurements_file = rospy.get_param('~measurements_file')
+        loop_rate = rospy.get_param('~loop_rate', 20)
 
         with open(measurements_file, 'r') as stream:
             try:
-                self.config_file = yaml.safe_load(stream)
+                self._config_file = yaml.safe_load(stream)
             except yaml.YAMLError as exc:
                 print(exc)
 
-        loop_rate = rospy.get_param('~loop_rate', 20)
-        rospy.Timer(rospy.Duration(1 / loop_rate), self._battery_callback)
+        self._V_driv_front = float('nan')
+        self._V_driv_rear = float('nan')
+        self._I_driv_front = float('nan')
+        self._I_driv_rear = float('nan')
 
-    def _battery_driv_callback(self, msg):
-        self.V_driv_front = msg.V_front
-        self.V_driv_rear = msg.V_rear
-        self.I_driv_front = msg.I_front
-        self.I_driv_rear = msg.I_rear
+        # -------------------------------
+        #   Publishers & Subscribers
+        # -------------------------------
 
-    def _battery_callback(self, *args):
-        try:
-            rospy.get_master().getPid()
-        except:
-            rospy.logerr(f'[{rospy.get_name()}] Error getting master')
-            exit(1)
+        self._battery_driv_sub = rospy.Subscriber('panther_driver/state', DriverState, self._battery_driv_cb)
+
+        self._battery_publisher = rospy.Publisher('battery', BatteryState, queue_size=1)
+        self._battery1_publisher = rospy.Publisher('battery1', BatteryState, queue_size=1)
+        self._battery2_publisher = rospy.Publisher('battery2', BatteryState, queue_size=1)
```

Comment:
```suggestion
        self._battery_1_publisher = rospy.Publisher('battery_1', BatteryState, queue_size=1)
        self._battery_2_publisher = rospy.Publisher('battery_2', BatteryState, queue_size=1)
```

I would do it this way